src/hotspot/share/gc/g1/g1CollectedHeap.cpp

```diff
@@ -2260,28 +2260,13 @@ bool G1CollectedHeap::try_collect(GCCause::Cause cause) {
 }
 
 bool G1CollectedHeap::is_in(const void* p) const {
-  if (_hrm->reserved().contains(p)) {
-    // Given that we know that p is in the reserved space,
-    // heap_region_containing() should successfully
-    // return the containing region.
-    HeapRegion* hr = heap_region_containing(p);
-    return hr->is_in(p);
-  } else {
-    return false;
-  }
-}
-
-#ifdef ASSERT
-bool G1CollectedHeap::is_in_exact(const void* p) const {
-  bool contains = reserved().contains(p);
-  bool available = _hrm->is_available(addr_to_region((HeapWord*)p));
-  if (contains && available) {
+  if (is_in_reserved(p) &&
+    _hrm->is_available(addr_to_region((HeapWord*)p))) {
```

Comment:
This line is misindented.  But rather than adjusting the indentation, how about instead changing the function to
```
bool G1CollectedHeap::is_in(const void* p) const {
  return is_in_reserved(p) && _hrm->is_available(addr_to_region((HeapWord*)p));
}
```

Reply:
I was thinking something similar, so I would approve going with this proposal.